### PR TITLE
implemented memory saving changes

### PR DESCRIPTION
**Pull request by jerryneedell**
No functional changes.

Made several changes to reduce the memory usage:
1) Removed use of "warnings"   for crc errors - replaced with a counter
2) Replaces some instances of "registerbits" with simple function calls ( rx_done,tx_done,crc_error are now functions and are only ever read so the setter/getter was not necessary)
3) made ACK packet a fixed byte string -- no need to call bytes to create it
4) remove unused constants

Tested on STM32F405 -- before an after changes
gc.mem_free() shows an improvement of ~2Kbytes 

before changes 
```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 5.4.0-beta.1-13-g9285252fb on 2020-06-12; Feather STM32F405 Express with STM32F405RG
>>> import gc
>>> gc.mem_free()
91088
>>> import adafruit_rfm9x
>>> gc.mem_free()
76112
>>> gc.collect()
>>> gc.mem_free()
76080
>>> 
```

after changes
```
Press any key to enter the REPL. Use CTRL-D to reload.
Adafruit CircuitPython 5.4.0-beta.1-13-g9285252fb on 2020-06-12; Feather STM32F405 Express with STM32F405RG
>>>
>>> import gc
>>> gc.mem_free()
91088
>>> import adafruit_rfm9x
>>> gc.mem_free()
79344
>>> gc.collect()
>>> gc.mem_free()
79312
>>> 
```

I have also tested "freezing" this into the feather_m0_rfm9x BSP CP build and It can be made to fit with some modules removed from the standard CP build -- once this PR is accepted, I will complete that testing and submit a PR for it.

There are no breaking changes in this PR unless a user was accessing self.rx_done,self.tx_done or self.crc_error  from user code. That is not done in any of the examples.

### CODE REVIEW

File: adafruit_rfm9x.py [hunk 1]
```diff
@@ -873,8 +752,8 @@ def receive(
         # Enter idle mode to stop receiving other packets.
         self.idle()
         if not timed_out:
-            if self.enable_crc and self.crc_error:
-                warn("CRC error, packet ignored")
+            if self.enable_crc and self.crc_error():
+                self.crc_error_count += 1
```
Comment (brentru) [inline adafruit_rfm9x.py:756]: @jerryneedell  q: with `warn` removed, user will not be notified of a CRC error. Would this be handled in user-code?

Comment (jerryneedell) [reply adafruit_rfm9x.py:756]: yes -- something like this 
```
crc_count = 0
while True:
    # Look for a new packet: only accept if addresses to my_node
    packet = rfm9x.receive(with_ack=True, with_header=True)
    if rfm9x.crc_error_count != crc_count:
        crc_count = rfm9x.crc_error_count
        print("CRC error ",crc_count)
```

Comment (jerryneedell) [reply adafruit_rfm9x.py:756]: In many cases -- user may not care or want to know -- the packet will just be ignored

File: adafruit_rfm9x.py [hunk 2]
```diff
@@ -108,118 +93,9 @@ def warn(msg, **kwargs):
...
-_RH_RF95_SPREADING_FACTOR_64CPS = const(0x60)
-_RH_RF95_SPREADING_FACTOR_128CPS = const(0x70)
-_RH_RF95_SPREADING_FACTOR_256CPS = const(0x80)
-_RH_RF95_SPREADING_FACTOR_512CPS = const(0x90)
-_RH_RF95_SPREADING_FACTOR_1024CPS = const(0xA0)
-_RH_RF95_SPREADING_FACTOR_2048CPS = const(0xB0)
-_RH_RF95_SPREADING_FACTOR_4096CPS = const(0xC0)
-_RH_RF95_TX_CONTINUOUS_MOE = const(0x08)
-_RH_RF95_AGC_AUTO_ON = const(0x04)
-_RH_RF95_SYM_TIMEOUT_MSB = const(0x03)
```
Comment (tannewt) [inline adafruit_rfm9x.py:220]: Are these unused const values kept in memory? If they are used, they are usually copied in place and the name dropped. Maybe we need to modify how const works.

File: adafruit_rfm9x.py [hunk 3]
```diff
@@ -721,6 +588,18 @@ def enable_crc(self, val):
...
+        """Transmit status"""
+        return (self._read_u8(_RH_RF95_REG_12_IRQ_FLAGS) & 0x8) >> 3
+
+    def rx_done(self):
+        """Receive status"""
+        return (self._read_u8(_RH_RF95_REG_12_IRQ_FLAGS) & 0x40) >> 6
+
+    def crc_error(self):
+        """crc status"""
+        return (self._read_u8(_RH_RF95_REG_12_IRQ_FLAGS) & 0x20) >> 5
```
Comment (tannewt) [inline adafruit_rfm9x.py:601]: How much memory does this save? Seems like it'd very close to the cost of three instances of `_RegisterBits`.

File: adafruit_rfm9x.py (hunk 2 above)
Comment (jerryneedell) [reply adafruit_rfm9x.py:220]: I'm not sure this made any difference. I took them out since they were not used.

File: adafruit_rfm9x.py (hunk 3 above)
Comment (jerryneedell) [reply adafruit_rfm9x.py:601]: The main saving is that register bits creates the getter and setter -- these are all read only so they are never set. still -- it's not huge savings.

Comment (tannewt) [reply adafruit_rfm9x.py:601]: The getter and setter functions should be shared though because it's three instances of the same class.

Comment (jerryneedell) [reply adafruit_rfm9x.py:601]: I don’t understand. There is no setter.

Comment (jerryneedell) [reply adafruit_rfm9x.py:601]: Are you saying that with registerbits the unused setter dies not cost anything?

Comment (tannewt) [reply adafruit_rfm9x.py:601]: Yes. The function objects in the data descriptor class (registerbits) should be shared amongst all instances if my mental model is correct.

**Pull request merged**